core/src/main/scala/org/ensime/indexer/stringymap/StringyMap.scala

```diff
@@ -116,9 +133,49 @@ package object impl {
     // HACK: really need a Wrapper like sjs
     // WORKAROUND: edge names cannot have dots in them
     def label: String = tpe.describe.replace(".type", "")
-    def toProperties(t: T): StringyMap = sg.value.toProperties(gen.to(t))
+    def toProperties(t: T): StringyMap = {
+      val map = sg.value.toProperties(gen.to(t))
+      map.put("typehint", label)
+      map
+    }
     def fromProperties(m: StringyMap): BigResult[T] =
       sg.value.fromProperties(m).right.map(gen.from)
+    def toSchema: Map[String, (OType, Boolean)] = sg.value.toSchema
+  }
+
+  implicit def CNilBigDataFormat[T]: BigDataFormat[CNil] = new BigDataFormat[CNil] {
+    override def label: String = ???
+    override def toProperties(t: CNil): StringyMap = ???
+    override def toSchema: Map[String, (OType, Boolean)] = ???
+    override def fromProperties(m: StringyMap): BigResult[CNil] = ???
+  }
+
+  implicit def CoproductBigDataFormat[Key <: Symbol, Value, Tail <: Coproduct](
```

Comment:
by adding the `label` onto the `HList` implementation I guess we don't really need a coproduct implementation because that makes it handle classes that aren't part of a sealed family. With that in mind, maybe we should do it properly and drop the `label` and only allow it to work with sealed families?